python-backend/app/api/mines/incidents/resources/mine_incidents.py

```diff
@@ -295,8 +305,33 @@ def put(self, mine_guid, mine_incident_guid):
                     db.session.delete(doc)
                     db.session.commit()
 
-        try:
-            incident.save()
-        except Exception as e:
-            raise InternalServerError(f'Error when saving: {e}')
+        incident.save()
         return incident
+
+    @classmethod
+    def _handle_recommendations(cls, incident, recommendations):
+        for payload in recommendations:
+            edited_rec_guid = payload.get('mine_incident_recommendation_guid')
+            rec_string = payload.get('recommendation')
+            if rec_string is None or rec_string == '':
+                if payload.get('mine_incident_recommendation_guid') is not None:
+                    # Remove existing empty recommendations
+                    rec = MineIncidentRecommendation.find_by_mine_incident_recommendation_guid(
+                        edited_rec_guid)
+                    rec.soft_delete()
```

Comment:
not sure this needs a method.

Reply:
I can remove it if we don't like it. Just seemed like a nice complement to `.create()`

Reply:
I get where you're coming from. Might be better served on the base model, or even as an override of delete which could set delete if the model has deleted_ind, else delete.

Reply:
Yea I was going down Ryan's path, if we want a global soft_delete() then is should exist on the base model, but not all models have delete_ind (unless we want that to be a Mixin).